include/AST/Types/StructTy.hpp

```diff
@@ -15,14 +18,39 @@ class FieldDecl;
 namespace glu::types {
 
 /// @brief StructTy is a class that represents structures declared in code.
-class StructTy final : public TypeBase {
+class StructTy final : public TypeBase,
```

Comment:
we need to make sure StructDecl->getType() isn't used anywhere where using a parameterized templated value is necessary